### PR TITLE
Priority_Queue.py

### LINKED ISSUE #970: Priority queue in Python
I am an RGSoC participant and would like to implement priority queue in python.

### PR DESCRIPTION
**Pull request by VIJAY07102**
fixes #970

Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Priority_Queue/priority_queue.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    Invalid option!
+    Try Again!!
+
+'''
+
+while 1:
+    opt=0                  # initializing option
+    print("\t1. Insert value in queue\n\t2. Delete value in queue\n\t3. Print Queue\n\t4. Exit")
+    opt=int(input("Enter your option:"))
+    if opt==1:
```
Comment (jainaman224) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like `==`, `=`, `+`.

File: Priority_Queue/priority_queue.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        2. Delete value in queue
+        3. Print Queue
+        4. Exit
+Enter your option:2          #To delete the value of highest priority(here highest value)
+14 is deleted
+
+Enter your option:3          #To display the queue
+[12, 1, 7]
+
+Enter your option:4           #to exit
```
Comment (jainaman224) [inline Priority_Queue/priority_queue.py]: Add space after `#`.

File: Priority_Queue/priority_queue.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        item = queue[max] 
+        del queue[max] 
+        return item 
+    except IndexError:              #Instead of giving error for empty queue
+        print("Queue is empty!!")   #Displays that 'Queue is empty'
+        exit()                      #Exits the program if trying to delete from empty queue
+    
+
+#The program is made of user choice
+'''
```
Comment (jainaman224) [inline Priority_Queue/priority_queue.py]: Move this to end of file.

**Comment by jainaman224** [comment]
Fix PR title.

**Comment by VIJAY07102** [comment]
@jainaman224 Can you review this pull request, so that one of my PR can be merged. I hope, I made all the changes to get it merged.

File: Priority_Queue/priority_queue.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+queue=[]
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add space around =

File: Priority_Queue/priority_queue.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        item = queue[max] 
+        del queue[max] 
+        return item 
+    except IndexError:              #Instead of giving error for empty queue
+        print("Queue is empty!!")   #Displays that 'Queue is empty'
+        exit()                      #Exits the program if trying to delete from empty queue
+    
+
+#The program is made of user choice
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Example should be at the end of the file

File: Priority_Queue/priority_queue.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
+queue=[]
+
+def insert(data):           #To insert values into the Queue
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add space after # 
Change it in the whole file

File: Priority_Queue/priority_queue.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+Enter your option:4           #to exit
+If you enter any other value other than (1,2,3,4), You will be shown:
+    Invalid option!
+    Try Again!!
+
+'''
+
+while 1:
+    opt=0                  # initializing option
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like ==, =, +.

File: Priority_Queue/priority_queue.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    print("\t1. Insert value in queue\n\t2. Delete value in queue\n\t3. Print Queue\n\t4. Exit")
+    opt=int(input("Enter your option:"))
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like ==, =, +.

File: Priority_Queue/priority_queue.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    if opt==1:
+        value=int(input("Enter the value:"))
+        insert(value)
+    elif opt==2:
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like ==, =, +.

File: Priority_Queue/priority_queue.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        del_value=delete()
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like ==, =, +.

File: Priority_Queue/priority_queue.py [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        print("\n"+str(del_value)+" is deleted")
+    elif opt==3:
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add spaces around operators like ==, =, +.

File: Priority_Queue/priority_queue.py [hunk 12]
```diff
@@ -0,0 +1,78 @@
+
+# The program is made of user choice
+'''
+Example:
+        1. Insert value in queue
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Sample should be at the end of the file

File: Priority_Queue/priority_queue.py [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+Enter your option:4           #to exit
+If you enter any other value other than (1,2,3,4), You will be shown:
+    Invalid option!
+    Try Again!!
+
+'''
+
+
+
+queue=[]
```
Comment (somya-kapoor) [inline Priority_Queue/priority_queue.py]: Add space around =

**Comment by somya-kapoor** [comment]
@VIJAY07102  Are you working on it?

**Comment by VIJAY07102** [comment]
@somya-Kapoor yeah, mam I have completed this one already. Can I I upload a new file rather than resolving the conversation?

**Comment by VIJAY07102** [comment]
> @VIJAY07102 Are you working on it?

@somya-kapoor Mam please review the 'update priority_queue.py'

**Comment by VIJAY07102** [comment]
@somya-kapoor  I think now the file is ready to be merged. Please review this soon.

**Comment by somya-kapoor** [comment]
@VIJAY07102 Please squash the commits

**Comment by somya-kapoor** [comment]
@VIJAY07102 Are you working on it?

**Comment by VIJAY07102** [comment]
> @VIJAY07102 Are you working on it?

Mam, I'm unable to squash all the commits, can I make a new pull request in which everything will be perfect enough to merge it.

**Comment by somya-kapoor** [comment]
> > @VIJAY07102 Are you working on it?
> 
> Mam, I'm unable to squash all the commits, can I make a new pull request in which everything will be perfect enough to merge it.

Okay

**Pull request closed without merging**